lib/features/base/base_mailbox_controller.dart

```diff
@@ -554,4 +611,22 @@ abstract class BaseMailboxController extends BaseController {
     mailboxNode = personalMailboxTree.value.findNodeOnFirstLevel((node) => node.item.name?.name.toLowerCase() == name);
     return mailboxNode;
   }
+
+  void copySubaddressAction(BuildContext context, String subaddress) {
+    Clipboard.setData(ClipboardData(text: subaddress));
+    appToast.showToastSuccessMessage(context, AppLocalizations.of(context).emailSubaddressCopiedToClipboard);
+  }
+
+  String getSubaddress(String userEmail, String folderName) {
+    if (folderName.isEmpty) {
+      throw EmptyFolderNameException(folderName);
+    }
+
+    final atIndex = userEmail.indexOf('@');
+    if (atIndex <= 0 || atIndex == userEmail.length - 1) {
+      throw const FormatException('Invalid email format');
```

Comment:
Should create class for it same with `EmptyFolderNameException `

Reply:
done !